src/analyses/ai_domain.h

```diff
@@ -137,6 +137,11 @@ class ai_domain_factory_baset
 
   virtual std::unique_ptr<statet> make_domain(locationt l) const = 0;
   virtual std::unique_ptr<statet> copy_domain(const statet &s) const = 0;
+
+  // Not domain construction but requires knowing the precise type of statet
+  virtual bool
+  merge(statet &dest, const statet &src, locationt from, locationt to)
```

Comment:
Rather than apologise for "factory" containing a non-factory method, how about "domain-manager"?

Reply:
Yes ... probably the right name for it but ... uhhh... that's going to be a massive pain to fix as it is in comments, commit messages and so on and for much of the patch set it is a sensible name.

I am going to open a discussion issue on renaming things as @peterschrammel requested, can we deal with this as part of that?